### PR TITLE
Allow to iterate a byte array inside the smart contract

### PR DESCRIPTION
**Pull request by shargon**
Close https://github.com/neo-project/neo/issues/1172
Wait for https://github.com/neo-project/neo/pull/1178

This code works if the StackItem is an Array, not a buffer, but I think that this is not the expected behaviour for a c# developer.

```csharp
public static int TestNext(byte[] a)
        {
            int sum = 0;
            var enumerator = Enumerator<byte>.Create(a);

            while (enumerator.Next())
            {
                sum += enumerator.Value;
            }

            return sum;
        }
```

### CODE REVIEW

File: neo/SmartContract/InteropService.NEO.cs [hunk 1]
```diff
@@ -259,11 +259,20 @@ private static bool Storage_Find(ApplicationEngine engine)
...
-                IEnumerator enumerator = new ArrayWrapper(array);
-                engine.CurrentContext.EvaluationStack.Push(StackItem.FromInterface(enumerator));
-                return true;
+                case VMArray array:
+                    {
+                        IEnumerator enumerator = new ArrayWrapper(array);
+                        engine.CurrentContext.EvaluationStack.Push(StackItem.FromInterface(enumerator));
+                        return true;
+                    }
+                case ByteArray buffer:
```
Comment (erikzhang) [inline neo/SmartContract/InteropService.NEO.cs]: I think this should be `PrimitiveType`. Please wait for #1178.

File: neo/SmartContract/Iterators/ArrayWrapper.cs [hunk 2]
```diff
@@ -14,9 +14,7 @@ public ArrayWrapper(IList<StackItem> array)
             this.array = array;
         }
 
-        public void Dispose()
-        {
-        }
+        public void Dispose() { }
```
Comment (erikzhang) [inline neo/SmartContract/Iterators/ArrayWrapper.cs]: No need to change this file...

File: neo/SmartContract/Iterators/BufferWrapper.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+using System;
+
+namespace Neo.SmartContract.Iterators
+{
+    internal class BufferWrapper : IIterator
+    {
+        private readonly byte[] array;
+        private int index = -1;
+
+        public BufferWrapper(ByteArray array)
```
Comment (erikzhang) [inline neo/SmartContract/Iterators/BufferWrapper.cs]: `PrimitiveType`

**Comment by shargon** [comment]
Moved to https://github.com/neo-project/neo/pull/1281

**Pull request closed without merging**